### PR TITLE
feat(Naxxramas25/Gluth): Add Flesh Rip warning and icons on target

### LINKED ISSUE #39: <4445> Gluth(25) 

New warning + Mark "Flesh Rip"  (Run little girl RUN aswell?)

~~will need 2 marks as the debuffs overlap by 5 sec,~~
Replacing the mark early is fine, just use 1 marker, OR if you can dont apply new marker before old debuff runs out

Debuff = flesh rip, ID 40199
<4445> Gluth(25) 

New warning + Mark "Flesh Rip"  (Run little girl RUN aswell?)

~~will need 2 marks as the debuffs overlap by 5 sec,~~
Replacing the mark early is fine, just use 1 marker, OR if you can dont apply new marker before old debuff runs out

Debuff = flesh rip, ID 40199  - raised by Holycoke via Discord link: https://discord.com/channels/808813760978223135/882719072763121685/884823493152227378

### LINKED ISSUE #40: <4445> Gluth(25) 

enrage timer is wrong, boss enrage after 4th Decimate, not on the current timer..

hack fix would be to increase the enrage timer by 1 min, should at least make it MORE accurate
<4445> Gluth(25) 

enrage timer is wrong, boss enrage after 4th Decimate, not on the current timer..

hack fix would be to increase the enrage timer by 1 min, should at least make it MORE accurate  - raised by Holycoke via Discord link: https://discord.com/channels/808813760978223135/882719072763121685/884825118872854568

### PR DESCRIPTION
**Pull request by Exeasd**
Closes #39 
Closes #40

### CODE REVIEW

File: DBM-Naxx/ConstructQuarter/Gluth.lua [hunk 1]
```diff
@@ -3,26 +3,39 @@ local L		= mod:GetLocalizedStrings()
...
+local fleshripIcons = {}
+
 function mod:OnCombatStart(delay)
-	enrageTimer:Start(420 - delay)
 	timerDecimate:Start(110 - delay)
 	warnDecimateSoon:Schedule(100 - delay)
+	if mod:IsDifficulty("heroic25") then
+		enrageTimer:Start(480 - delay)
+	else
+	enrageTimer:Start(420 - delay)
```
Comment (Siarkowy) [inline DBM-Naxx/ConstructQuarter/Gluth.lua:37]: Uhh missed indent

Comment (Exeasd) [reply DBM-Naxx/ConstructQuarter/Gluth.lua:37]: why would you comment on stuff that is already in the core?! all i did is copy paste to fix the differences that didnt let us squash and merge

File: DBM-Naxx/ConstructQuarter/Gluth.lua [hunk 2]
```diff
@@ -3,24 +3,37 @@ local L		= mod:GetLocalizedStrings()
...
+local enrageTimer		= mod:NewBerserkTimer(480)
 local timerDecimate		= mod:NewCDTimer(104, 54426)
 
+mod:AddBoolOption("SetIconOnFleshRipTarget", true)
+
+local fleshripIcons = {}
+
 function mod:OnCombatStart(delay)
-	enrageTimer:Start(420 - delay)
+	if mod:IsDifficulty("heroic25") then
```
[the remark this answers is not in the document]
Comment (Exeasd) [reply DBM-Naxx/ConstructQuarter/Gluth.lua:34]: same as the other one

File: DBM-Naxx/ConstructQuarter/Gluth.lua [hunk 3]
```diff
@@ -3,24 +3,37 @@ local L		= mod:GetLocalizedStrings()
 
 mod:SetRevision(("$Revision: 2869 $"):sub(12, -3))
 mod:SetCreatureID(15932)
+mod:SetUsedIcons(7, 8)
```
[the remark this answers is not in the document]
Comment (Exeasd) [reply DBM-Naxx/ConstructQuarter/Gluth.lua:6]: there can be maximum of 2 targets, so theres rly no need for us to use more than 2 emotes

File: DBM-Naxx/ConstructQuarter/Gluth.lua [hunk 4]
```diff
@@ -3,24 +3,37 @@ local L		= mod:GetLocalizedStrings()
...
 )
 
 
 local warnDecimateSoon	= mod:NewSoonAnnounce(54426, 2)
 local warnDecimateNow	= mod:NewSpellAnnounce(54426, 3)
+local specwarnFleshRip	= mod:NewSpecialWarning("SpecialWarningFleshRip")
+local warnFleshRip		= mod:NewTargetAnnounce(40199, 2)
 
-local enrageTimer		= mod:NewBerserkTimer(420)
+local enrageTimer		= mod:NewBerserkTimer(480)
```
[the remark this answers is not in the document]
Comment (Exeasd) [reply DBM-Naxx/ConstructQuarter/Gluth.lua]: but but i know, thats why its outdated :<

File: DBM-Naxx/ConstructQuarter/Gluth.lua [hunk 5]
```diff
@@ -39,4 +49,45 @@ function mod:SPELL_DAMAGE(args)
...
+function mod:SPELL_AURA_REMOVED(args)
+	if args:IsSpellID(40199) and self.Options.SetIconOnInjectionTarget then
+		removeIcon(args.destName)
+	end
+end
+
+local function addIcon()
+	for i,j in ipairs(fleshripIcons) do
+		local icon = 9 - i
+		self:SetIcon(j, icon)
```
Comment (Siarkowy) [inline DBM-Naxx/ConstructQuarter/Gluth.lua:74]: self -> mod (there is no self here, cause function is not called on object)

File: DBM-Naxx/ConstructQuarter/Gluth.lua [hunk 6]
```diff
@@ -39,4 +49,45 @@ function mod:SPELL_DAMAGE(args)
...
+	end
+end
+
+local function removeIcon(target)
+	for i,j in ipairs(fleshripIcons) do
+		if j == target then
+			table.remove(fleshripIcons, i)
+			self:SetIcon(target, 0)
```
Comment (Siarkowy) [inline DBM-Naxx/ConstructQuarter/Gluth.lua:82]: self -> mod

File: DBM-Naxx/ConstructQuarter/Gluth.lua [hunk 7]
```diff
@@ -3,29 +3,39 @@ local L		= mod:GetLocalizedStrings()
...
+mod:AddBoolOption("SetIconOnFleshRipTarget", true)
+
+local fleshripIcons = {}
+
 function mod:OnCombatStart(delay)
 	timerDecimate:Start(110 - delay)
 	warnDecimateSoon:Schedule(100 - delay)
-	if mod:IsDifficulty("heroic25") then
+
+	if self:IsDifficulty("heroic25") then
```
Comment (Siarkowy) [inline DBM-Naxx/ConstructQuarter/Gluth.lua:35]: Good (function called on "mod" object)

File: DBM-Naxx/ConstructQuarter/Gluth.lua [hunk 8]
```diff
@@ -39,4 +49,45 @@ function mod:SPELL_DAMAGE(args)
...
+			table.remove(fleshripIcons, i)
+			self:SetIcon(target, 0)
+		end
+	end
+	addIcon()
+end
+
+function mod:OnCombatEnd()
+    for i,j in ipairs(fleshripIcons) do
+       self:SetIcon(j, 0)
```
Comment (Siarkowy) [inline DBM-Naxx/ConstructQuarter/Gluth.lua:90]: Good (self points to "mod" object)

**Comment by Siarkowy** [comment]
Whenever a function is declared/called with colon :, it has implicit "self" variable.

Basically colon call is just syntax sugar: mod:function(...) == mod.func(mod, ...). It is commonly used to mimic method calls, although there's no classes/objects in Lua; only prototypes (and metatables "blessing" a table to become an "object").

**Review by Siarkowy — APPROVED** [review]
Very nice!!

So now when you look at it, is it "self" explanatory? 😛

**Pull request merged**